### PR TITLE
User migration - clean up legacy data

### PR DESCRIPTION
**Pull request by Gio2018**
## Summary
* This PR cleans up old legacy data once the user is migrated
> **Note**
this PR should not be merged until the migration PR gets merged
- See issue number referenced in the branch name

## Implementation Details
* Add `LegacyCleanupService`, a service responsible of removing legacy data:
    - sqlite db
    - log files
    - image cache
    - audio cache
    - `UserDefaults`

## Test Steps
NA

## PR Checklist:
- [ ] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [ ] Basic Self QA

### CODE REVIEW

**Review by nzeltzer — APPROVED** [review]
lgtm

File: PocketKit/Sources/SharedPocketKit/Migrations/Cleanup/CleanupService.swift [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    case fileNotFound
+}
+
+/// Service that cleans up legacy data
+public struct LegacyCleanupService {
+    public func cleanUp() {
+        cleanupLegacyUserDefaults()
+        deleteSqliteDataBase()
+        deleteImageCache()
+        deleteLogFiles()
```
Comment (cyndichin) [inline PocketKit/Sources/SharedPocketKit/Migrations/Cleanup/CleanupService.swift:17]: should we also add `deleteAudioCache` here as well? Or will it be handled somewhere else, if so why?

Comment (Gio2018) [reply PocketKit/Sources/SharedPocketKit/Migrations/Cleanup/CleanupService.swift]: indeed we should @cyndichin , thanks for catching this!

Comment (cyndichin) [reply PocketKit/Sources/SharedPocketKit/Migrations/Cleanup/CleanupService.swift]: np! seems like you have 2 approvals already, but can re-review when conflicts addressed

**Pull request merged**